### PR TITLE
Move to kotlin

### PR DESCRIPTION
**Pull request by hwbllmnn**
@terrestris/devs Please review.

### CODE REVIEW

File: src/main/kotlin/de/terrestris/keycloak/ldap/LdapDnMapper.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  ): List<UserModel> {
+    return super.getRoleMembers(realm, role, firstResult, maxResults)
+  }
+
+  init {
+    log.debug("Using index $index")
+    log.debug("Creating groups: " + if (createGroups) "yes" else "no")
+  }
+
+}
```
Comment (simonseyock) [inline src/main/kotlin/de/terrestris/keycloak/ldap/LdapDnMapper.kt:91]:
```suggestion
}

```

File: src/main/kotlin/de/terrestris/keycloak/ldap/LdapDnMapper.kt [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        continue
+      }
+      // dns may apparently contain multiple values per step like uid=john+sn=Doe, but we're ignoring this for now
+      val key = rdn.allKeys[0]
+      dnPath.computeIfAbsent(key) { k: String? -> ArrayList() }
+      dnPath[key]!!.add(rdn.getAttrValue(key))
+    } while (dn.parentDn.also { dn = it } != null && !dn.toString().isEmpty())
+    val part = dnPath["ou"]!![index]
+    if (createGroups) {
+      val groups = realm.groupsStream.filter { g: GroupModel -> g.name == part }.collect(Collectors.toList())
```
Comment (simonseyock) [inline src/main/kotlin/de/terrestris/keycloak/ldap/LdapDnMapper.kt:44]: normally a sequence would be preferred over a stream, but i suppose that this is not convenient here

File: src/main/kotlin/de/terrestris/keycloak/ldap/LdapMapperFactory.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  }
+
+  override fun getTypeMetadata(): Map<String, Any> {
+    val metadata: MutableMap<String, Any> = HashMap()
+    metadata["fedToKeycloakSyncSupported"] = true
+    metadata["keycloakToFedSyncSupported"] = false
+    return metadata
+  }
+
+}
```
Comment (dnlkoch) [inline src/main/kotlin/de/terrestris/keycloak/ldap/LdapMapperFactory.kt:55]:
```suggestion
}

```

**Review by dnlkoch — APPROVED** [review]

File: src/main/kotlin/de/terrestris/keycloak/ldap/LdapMapperFactory.kt (hunk 3 above)
Comment (simonseyock) [inline src/main/kotlin/de/terrestris/keycloak/ldap/LdapMapperFactory.kt:55]:
```suggestion
}

```

**Review by simonseyock — APPROVED** [review]
very nice!

File: src/main/kotlin/de/terrestris/keycloak/ldap/LdapDnMapper.kt (hunk 2 above)
Comment (hwbllmnn) [reply src/main/kotlin/de/terrestris/keycloak/ldap/LdapDnMapper.kt:44]: The code was auto-converted via IntelliJ, I only adjusted parts that wouldn't compile.

**Pull request merged**